service/src/main/java/org/jboss/sbomer/service/feature/sbom/model/SbomGenerationRequest.java

```diff
@@ -0,0 +1,87 @@
+package org.jboss.sbomer.service.feature.sbom.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.Table;
+import javax.transaction.Transactional;
+
+import org.hibernate.annotations.DynamicUpdate;
+import org.hibernate.annotations.TypeDef;
+import org.jboss.sbomer.service.feature.sbom.k8s.model.GenerationRequest;
+import org.jboss.sbomer.service.feature.sbom.k8s.model.SbomGenerationStatus;
+
+import io.quarkiverse.hibernate.types.json.JsonBinaryType;
+import io.quarkiverse.hibernate.types.json.JsonTypes;
+import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+@DynamicUpdate
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = true)
+@Entity
+@TypeDef(name = JsonTypes.JSON_BIN, typeClass = JsonBinaryType.class)
+@ToString
+@Table(
+		name = "sbom_generation_request",
+		indexes = { @Index(name = "idx_request_buildid", columnList = "build_id"),
+				@Index(name = "idx_request_status", columnList = "status") })
+@Slf4j
+public class SbomGenerationRequest extends PanacheEntityBase {
+
+	@Id
+	@Column(nullable = false, updatable = false)
+	private String id;
+
+	@Column(name = "status", nullable = false)
+	@Enumerated(EnumType.STRING)
+	SbomGenerationStatus status;
+
+	@Column(name = "build_id", nullable = false, updatable = false)
```

Comment:
Good catch, but: it doesn't fail, because it is added only at the creation time. Even when we are running the method again, the setter is a noop, since the buildId is the same.

Reply:
I'll update the code to show it better.